src/utils.ts

```diff
@@ -206,3 +206,46 @@ export const addTokenEventListener = async (
     contractAddress,
     ClientType.TokenClient,
   );
+
+/**
+ * Extension specific event listener
+ * It creates a new interface with ABI data describing
+ * possible event signature
+ */
+export const addExtensionEventListener = async (
+  eventSignature: ContractEventsSignatures,
+  extensionAddress: string,
+): Promise<void> => {
+  const { provider } = networkClient;
+  const iface = new utils.Interface([`event ${eventSignature}`]);
+  const filter = {
+    topics: [utils.id(eventSignature)],
+    address: extensionAddress,
+  };
+
+  verbose(
+    'Added listener for Event:',
+    eventSignature,
+    extensionAddress ? `filtering Address: ${extensionAddress}` : '',
+  );
+
+  provider.on(filter, async (log: Log) => {
```

Comment:
Do you know why using the networkClient interface errors when parsing the log for `ExtensionInitialised()`? I'm trying to figure it out. 

If it's because `ExtensionInitialised` is not a network-level event and therefore not in the network abi, then how come the event still gets picked up here by the network client provider?

(I'm presuming that's the reason for `addExtensionEventListener` instead of using `eventListenerGenerator`?)

Reply:
I have no idea. As you mentioned, `parseLog` was erroring when using the networkClient interface and I had to add the corresponding ABI to the custom interface. 
@rdig might be able to shine more light here

Reply:
Yeah, basically that it. The event is not present on the network contract so it doesn't know how to parse it. _(this is also because of Ethers v5 which throws instead of just returning null like the old v4)_

Jakub's solution works since he basically modified the contract _(from Ether's perspective)_